### PR TITLE
Gen 3 Randbat improvements

### PR DESCRIPTION
**Pull request by Honko**
* Reject Mean Look, Spider Web, and Protect without moves to support them
* Reject Sunny Day without Solar Beam and vice versa
* Reject Reversal/Endeavor without Sub/Endure
* Reject multiple self-KO moves on the same set
* Reject Whirlwind with setup
* Reject Endure with Substitute
* Reject redundant Grass- and Fighting-type moves
* Remove checks for moves that don't exist or are irrelevant in ADV
* Fix bug that sometimes caused Castform to have its weather move twice

### CODE REVIEW

**Comment by TheImmortal** [comment]
@HoeenCoder

File: mods/gen3/random-teams.js [hunk 1]
```diff
@@ -98,46 +98,56 @@ class RandomGen3Teams extends RandomGen4Teams {
...
 					if (counter.setupType || !!counter['recovery'] || hasMove['rest']) rejected = true;
+					if (moveid === 'destinybond' && (hasMove['explosion'] || hasMove['selfdestruct'])) rejected = true;
+					if (moveid === 'memento' && hasMove['destinybond']) rejected = true;
 					break;
-				case 'foresight': case 'protect': case 'roar':
+				case 'protect':
+					if (!hasAbility['Speed Boost'] && !hasMove['perishsong'] && !hasMove['toxic'] && !hasMove['wish']) rejected = true;
+					break;
+				case 'roar': case 'whirlwind':
 					if (counter.setupType && !hasAbility['Speed Boost']) rejected = true;
```
Comment (HoeenCoder) [inline mods/gen3/random-teams.js]: Not sure why having Speed boost would help roar/whirlwind as they are both -6 priority.

Comment (Honko) [reply mods/gen3/random-teams.js]: Good catch, I missed taking that out when I split Protect into its own check.

**Comment by HoeenCoder** [comment]
Looks good after testing.

**Comment by HoeenCoder** [comment]
Whoops almost forgot, tagging @TheImmortal to merge

**Pull request merged**